homework3/hw3/ViewModel/EventsViewModel.swift

```diff
@@ -0,0 +1,30 @@
+//
+//  EventsViewModel.swift
+//  hw3
+//
+//  Created by Tina Jureško on 27.03.2025..
+//
+
+import Foundation
+import UIKit
+import SofaAcademic
+import SnapKit
+
+struct EventsViewModel {
+    var sections: [LeagueSection] = []
+    
+    func getLeagueSections(allEvents: [Event]) -> [LeagueSection] {
+        
+        let grouped = Dictionary(grouping: allEvents, by: { $0.league?.id })
+        
+        return grouped.compactMap { (key, events) in
+            guard let league = events.first?.league else {
+                return nil
+            }
```

Comment:
Hoće ovo oba puta uzeti prvi element ligu?
Kako to da nisi napravila po keyevima iteraciju?